account_mass_reconcile/models/base_reconciliation.py

```diff
@@ -158,9 +158,20 @@ def create_write_off(self, lines, amount, amount_curr, same_curr):
         currency = same_curr and lines[0].currency_id or lines[0].company_id.currency_id
         journal = self.journal_id
         partners = lines.mapped("partner_id")
+
+        # Adjust amount_currency to match the sign of the company's residual (amount)
+        if same_curr:
+            if amount != 0:
+                sign = 1 if amount > 0 else -1
+                adjusted_amount_curr = abs(amount_curr) * sign
+            else:
+                adjusted_amount_curr = 0.0
+        else:
+            adjusted_amount_curr = amount
+
         write_off_vals = {
             "name": _("Automatic writeoff"),
-            "amount_currency": same_curr and amount_curr or amount,
+            "amount_currency": adjusted_amount_curr if same_curr else amount,
```

Comment:
Can't you just take `adjusted_amount_curr` here because `adjusted_amount_curr` already contains the right value depending on the value of `same_curr`, making the condition here redundant (and confusing)?

Reply:
Done, thanks!